### DISCUSSION TITLE
Frequently getting "Failure (cannot unmarshal update response [...]" with NameCheap

### DISCUSSION THREAD

**Discussion by aclbandit** [qdm12/ddns-updater#265] [Q&A]
Frequently receiving "Failure (cannot unmarshal update response: xml: encoding "utf-16" declared but Decoder.CharsetReader is nil)" when updating my domains on NameCheap. When _not_ receiving that error, the domains update just fine. 

This could be a bug, but it also could be a configuration issue on my end. 

Any ideas?

Thanks!
![image](https://user-images.githubusercontent.com/38426796/142503723-79bd6cb3-372a-4c36-99d5-0a41655b5e3a.png)

**Comment by qdm12** [comment]
I personally use Namecheap and never had the error, that's strange.

Anyway, I added some workaround fix which should do it in 87ffce17b1d915fe8045f5bacd2ef0860a5a9d39

**Reply by pheetr** [reply]
I've also been getting these errors for the past few days. I had been using about a year old version of DDNS Updater without issues up until an upgrade to the latest version last week and the errors started popping up thereafter, so I thought it may have been related. In an effort to get to a clean state I've removed all the old data and updated the config with some of the new parameters, but these errors keep coming. 

As an example of what the experience is in my case, I got a notification from Uptime Kuma today at 3am that the service went down with status code 500. Ever since then the container has been in an 'unhealthy' state. Interestingly, when logging into Namecheap I can see that the IP was in fact updated successfully. 
The status page still shows the failure as the current status.
![image](https://user-images.githubusercontent.com/44616229/142773109-eda18360-c345-4332-8d8b-09cb541680e7.png)

Thx. :)

**Reply by qdm12** [reply]
That's most likely an error related to Namecheap I'd say. If you check [the commits logs](https://github.com/qdm12/ddns-updater/commits/master), there is not really any code-related commit in the last 30 days, except the fix mentioned above.

@pheetr try pulling the latest image, it might fix it. And yes it's just the program cannot verify the response from Namecheap, but the request might had been successful, especially since the HTTP status code is OK.

**Reply by pheetr** [reply]
No issues over the past two days or so, since I've updated the container, so I'd say the fix is working well.
Thanks again! :)

**Reply by qdm12** [reply]
Thanks for letting me know!
